### ISSUE TITLE
Regions plugin issues

### ISSUE THREAD

**Issue by brollin** [katspaugh/wavesurfer.ts#75]
I've been playing around with the regions plugin, and I can do some things successfully:
* create and register the plugin
* add regions programmatically
* create new regions with enableDragSelection
* listened successfully for "region-created" events 

Butt I've also had these problems:
* unable to listen to events "region-updated", "region-clicked"
* can't drag the edges of regions to resize them, even when passing `resize:true` (although the example didn't seem to need that for it to work 🤔)

Happy to provide more info if helpful.

**Comment by katspaugh** [comment]
Could you paste or link to the code that you tried?
Subscribing to `region-updated` and `region-clicked` should be no different than `region-created`.

Regarding the resizing, it's indeed enabled by default. I assume it's for Conjurer, which seems to have a pretty complex UI, so it might be a problem with DOM element layering. Something might be intercepting mousemove events is my best guess.

**Comment by brollin** [comment]
Thanks, that pointed me in the right direction and things are working now. I was creating a repro case to show you, and noticed I had changed the timeline plugin z-index to 10. When I remove that, things are working: https://github.com/SotSF/conjurer/blob/2a760a8b0483f1f832ecd73f9b81cdd54fe3bd2a/src/components/WavesurferWaveform.tsx#L42

Hopefully it's ok if I piggyback a couple of type issues I noticed:
* In typescript when applying styles to the timeline plugin, it gets mad that you aren't supplying all CSSStyleDeclaration keys. Maybe we want `Partial<CSSStyleDeclaration>` there.
* Somewhat similarly, when using the regions plugin `enableDragSelection`, It gets mad unless you provide `start` because that is required in `RegionParams`. Maybe it should be optional in that case.

In any case, my main issue is solved so thank you!
